analysis/interface/Events_2016v4.h

```diff
@@ -2071,8 +2071,12 @@ Events::~Events()
 {
    if (!fChain) return;
    delete fChain->GetCurrentFile();
-   if (h_fChain) delete h_fChain->GetCurrentFile();
-   if (ht_fChain) delete ht_fChain->GetCurrentFile();
+   if (!h_fChain) {
```

Comment:
I really don't understand what you're trying to do here?

Reply:
It was for solving segmentation violation when running Batch.py: 

The lines below might hint at the cause of the crash.
You may get help by asking at the ROOT forum http://root.cern.ch/forum.
Only if you are really convinced it is a bug in ROOT then please submit a
report at http://root.cern.ch/bugs. Please post the ENTIRE stack trace
from above as an attachment in addition to anything else
that might help us fixing this issue.
===========================================================
#6  0x00002ba3c1079f26 in TTree::GetCurrentFile() const () from /cvmfs/cms.cern.ch/slc6_amd64_gcc630/cms/cmssw/CMSSW_9_4_4/external/slc6_amd64_gcc630/lib/libTree.so
#7  0x00002ba3c0a157ee in Events::~Events (this=0x7ffc13931d90, __in_chrg=<optimized out>) at /cms/ldap_home/wjjang/wj_nanoAOD_CMSSW_9_4_4/src/nano/analysis/interface/Events_2016v4.h:2075
#8  0x0000000000403f57 in main ()
===========================================================

At first I have modified the code like below : 
   if (!fChain) return;
   delete fChain->GetCurrentFile();
   delete h_fChain->GetCurrentFile(); 
   delete ht_fChain->GetCurrentFile();

but I also saw the same error and then modified code like now ... 
I also don't understand why the error does not occur when the code is the same as it is now.